packages/dds/tree/src/feature-libraries/editable-tree/editableTree.ts

```diff
@@ -711,15 +718,26 @@ export class NodeProxyTarget extends ProxyTarget<Anchor> {
 		eventName: K,
 		listener: EditableTreeEvents[K],
 	): () => void {
-		assert(eventName === "changing", "unexpected eventName");
-		const unsubscribeFromValueChange = this.anchorNode.on("valueChanging", () => listener());
-		const unsubscribeFromChildrenChange = this.anchorNode.on("childrenChanging", () =>
-			listener(),
-		);
-		return () => {
-			unsubscribeFromValueChange();
-			unsubscribeFromChildrenChange();
-		};
+		// assert(eventName === "changing", "unexpected eventName");
+		if (eventName === "changing") {
```

Comment:
a better pattern here is to use a switch statement. Then in the default case call unreachableCase. This allows the compiler to give you an error if you are missing a case. Otherwise these changes look great.

Reply:
I know you marked this never merge, but with that fix, this would be a great change to merge. Seems quite useful!

Reply:
For sure the suggested is the right pattern. Editable-tree changes were more like a quick & dirty support for the higher-level binding ideas wanted to verify and validate with the community. Definitely, if appealing to the team I could make a dedicated PR with the additional `subtreeChanging` event in EditableTree.